ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java

```diff
@@ -157,7 +238,48 @@ public Object handle(Request request, Response response) throws Exception {
       return ImmutableMap.of("message", "Service not found");
     }
 
-    CqlQueryResponse cqlQueryResponse = cqlQueryUtil.executeCqlQuery(cqlRequest);
+    List<Result> results =
+        cqlRequests
+            .stream()
+            .map(
+                cqlRequest -> {
+                  CqlQueryResponse cqlQueryResponse = null;
+                  try {
+                    cqlQueryResponse = cqlQueryUtil.executeCqlQuery(cqlRequest);
+                  } catch (Exception e) {
+                    LOGGER.debug("Error fetching cql request for {}", cqlRequest.getSrc());
+                    return null;
+                  }
+                  return cqlQueryResponse.getQueryResponse().getResults();
+                })
+            .filter(cqlResults -> CollectionUtils.isNotEmpty(cqlResults))
+            .flatMap(cqlResults -> cqlResults.stream())
+            .collect(Collectors.toList());
+
+    results.sort(getResultComparators(cqlTransformRequest.getSorts()));
+
+    results =
+        results.size() > cqlTransformRequest.getCount()
+            ? results.subList(0, cqlTransformRequest.getCount())
+            : results;
+
+    results =
+        CollectionUtils.isEmpty(cqlTransformRequest.getHiddenResults())
+            ? results
+            : results
+                .stream()
+                .filter(
+                    result ->
+                        !cqlTransformRequest
+                            .getHiddenResults()
+                            .contains(result.getMetacard().getId()))
+                .collect(Collectors.toList());
+
+    QueryResponse combinedResponse =
+        new QueryResponseImpl(
+            new QueryRequestImpl(new QueryImpl(ECQL.toFilter(cqlRequests.get(0).getCql()))),
```

Comment:
is it possible for `cqlRequests` to be empty?

Reply:
No, I have a check for that here https://github.com/codice/ddf-ui/pull/105/files#diff-4ff7d3773f55434d06f165c2a670d3edR215